routes/web/project/vm.rb

```diff
@@ -45,7 +45,8 @@ class CloverWeb
         # that users won't see higher price in their invoice compared
         # to price calculator and also we charge same amount no matter
         # the number of days in a given month.
-        @prices = BillingRate.rates.each_with_object(Hash.new { |h, k| h[k] = h.class.new(&h.default_proc) }) do |br, hash|
+        @prices = BillingRate.rates.filter { ["VmCores", "IPAddress"].include?(_1["resource_type"]) }
```

Comment:
```suggestion
        types = ["VmCores", "IPAddress"].freeze
        @prices = BillingRate.rates.filter { types.include?(_1["resource_type"]) }
```

May be a dumb idea, may not be faster, but here's me running a disassembly to entertain myself on the subject:

```ruby
[5] clover-development(main)> RubyVM::InstructionSequence.compile('[1, 2].filter { [2].include? _1 }').to_a
=> ["YARVInstructionSequence/SimpleDataFormat",
 3,
 2,
 1,
 {:arg_size=>0, :local_size=>0, :stack_max=>1, :node_id=>13, :code_location=>[1, 0, 1, 33], :node_ids=>[1, 4, -1]},
 "<compiled>",
 "<compiled>",
 "<compiled>",
 1,
 :top,
 [],
 {},
 [],
 [1,
  :RUBY_EVENT_LINE,
  [:duparray, [1, 2]],
  [:send,
   {:mid=>:filter, :flag=>0, :orig_argc=>0},
   ["YARVInstructionSequence/SimpleDataFormat",
    3,
    2,
    1,
    {:arg_size=>1, :local_size=>1, :stack_max=>2, :node_id=>11, :code_location=>[1, 14, 1, 33], :node_ids=>[6, 7, 9, -1]},
    "block in <compiled>",
    "<compiled>",
    "<compiled>",
    1,
    :block,
    [:_1],
    {:lead_num=>1, :ambiguous_param0=>true},
    [],
    [1,
     :RUBY_EVENT_LINE,
     :RUBY_EVENT_B_CALL,
     [:duparray, [2]],
     [:getlocal_WC_0, 3],
     [:opt_send_without_block, {:mid=>:include?, :flag=>16, :orig_argc=>1}],
     :RUBY_EVENT_B_RETURN,
     [:leave]]]],
  [:leave]]]
[6] clover-development(main)> RubyVM::InstructionSequence.compile('selector = [2].freeze; [1, 2].filter { selector.include? _1 }').to_a
=> ["YARVInstructionSequence/SimpleDataFormat",
 3,
 2,
 1,
 {:arg_size=>0, :local_size=>1, :stack_max=>1, :node_id=>18, :code_location=>[1, 0, 1, 61], :node_ids=>[2, 3, 0, 5, 8, -1]},
 "<compiled>",
 "<compiled>",
 "<compiled>",
 1,
 :top,
 [:selector],
 {},
 [],
 [1,
  :RUBY_EVENT_LINE,
  [:duparray, [2]],
  [:opt_send_without_block, {:mid=>:freeze, :flag=>16, :orig_argc=>0}],
  [:setlocal_WC_0, 3],
  [:duparray, [1, 2]],
  [:send,
   {:mid=>:filter, :flag=>0, :orig_argc=>0},
   ["YARVInstructionSequence/SimpleDataFormat",
    3,
    2,
    1,
    {:arg_size=>1, :local_size=>1, :stack_max=>2, :node_id=>14, :code_location=>[1, 37, 1, 61], :node_ids=>[9, 10, 12, -1]},
    "block in <compiled>",
    "<compiled>",
    "<compiled>",
    1,
    :block,
    [:_1],
    {:lead_num=>1, :ambiguous_param0=>true},
    [],
    [1,
     :RUBY_EVENT_LINE,
     :RUBY_EVENT_B_CALL,
     [:getlocal_WC_1, 3],
     [:getlocal_WC_0, 3],
     [:opt_send_without_block, {:mid=>:include?, :flag=>16, :orig_argc=>1}],
     :RUBY_EVENT_B_RETURN,
     [:leave]]]],
  [:leave]]]
```